From 03f1540234207618c749f17a5a3b8fe6f8b1b09b Mon Sep 17 00:00:00 2001
From: Ikey Doherty <michael.i.doherty@intel.com>
Date: Wed, 24 Jun 2015 13:07:14 +0100
Subject: [PATCH 1/4] pam_securetty: Do not report non-fatal, documented,
 behaviour

Signed-off-by: Ikey Doherty <michael.i.doherty@intel.com>
---
 modules/pam_securetty/pam_securetty.c | 1 -
 1 file changed, 1 deletion(-)

diff --git a/modules/pam_securetty/pam_securetty.c b/modules/pam_securetty/pam_securetty.c
index 0474130..cbdb08f 100644
--- a/modules/pam_securetty/pam_securetty.c
+++ b/modules/pam_securetty/pam_securetty.c
@@ -111,7 +111,6 @@ securetty_perform_check (pam_handle_t *pamh, int ctrl,
     }
 
     if (stat(SECURETTY_FILE, &ttyfileinfo)) {
-	pam_syslog(pamh, LOG_NOTICE, "Couldn't open %s: %m", SECURETTY_FILE);
 	return PAM_SUCCESS; /* for compatibility with old securetty handling,
 			       this needs to succeed.  But we still log the
 			       error. */
-- 
2.7.1

