From 5d11a80b0477af8851c16eb25425c05c0d60ff0f Mon Sep 17 00:00:00 2001
From: William Douglas <william.douglas@intel.com>
Date: Thu, 29 Jun 2023 15:15:24 -0700
Subject: [PATCH 1/4] stateless move config dist dir

---
 libpam/pam_private.h | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/libpam/pam_private.h b/libpam/pam_private.h
index f9dd622..35c03fc 100644
--- a/libpam/pam_private.h
+++ b/libpam/pam_private.h
@@ -28,8 +28,8 @@
 #define PAM_CONFIG         "/etc/pam.conf"
 #define PAM_CONFIG_D       "/etc/pam.d"
 #define PAM_CONFIG_DF      "/etc/pam.d/%s"
-#define PAM_CONFIG_DIST_D  "/usr/lib/pam.d"
-#define PAM_CONFIG_DIST_DF "/usr/lib/pam.d/%s"
+#define PAM_CONFIG_DIST_D  "/usr/share/pam.d"
+#define PAM_CONFIG_DIST_DF "/usr/share/pam.d/%s"
 #ifdef VENDORDIR
 #define PAM_CONFIG_DIST2_D  VENDORDIR"/pam.d"
 #define PAM_CONFIG_DIST2_DF VENDORDIR"/pam.d/%s"
-- 
2.43.0

