From b8df38d55a2b811a52d4c6edcd025b2da8f10d9b Mon Sep 17 00:00:00 2001
From: William Douglas <william.douglas@intel.com>
Date: Thu, 29 Jun 2023 15:15:24 -0700
Subject: [PATCH 1/6] stateless move config dist dir

---
 libpam/pam_private.h | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/libpam/pam_private.h b/libpam/pam_private.h
index 508527c..9bddce7 100644
--- a/libpam/pam_private.h
+++ b/libpam/pam_private.h
@@ -27,8 +27,8 @@
 #define PAM_CONFIG         "/etc/pam.conf"
 #define PAM_CONFIG_D       "/etc/pam.d"
 #define PAM_CONFIG_DF      "/etc/pam.d/%s"
-#define PAM_CONFIG_DIST_D  "/usr/lib/pam.d"
-#define PAM_CONFIG_DIST_DF "/usr/lib/pam.d/%s"
+#define PAM_CONFIG_DIST_D  "/usr/share/pam.d"
+#define PAM_CONFIG_DIST_DF "/usr/share/pam.d/%s"
 #ifdef VENDORDIR
 #define PAM_CONFIG_DIST2_D  VENDORDIR"/pam.d"
 #define PAM_CONFIG_DIST2_DF VENDORDIR"/pam.d/%s"
-- 
2.41.0

