From 33c43e8e2f5301bd662219136037eba67ac3e112 Mon Sep 17 00:00:00 2001
From: William Douglas <william.douglas@intel.com>
Date: Thu, 29 Jun 2023 15:20:34 -0700
Subject: [PATCH] pam_env: Only report non-ENOENT errors for env file

Signed-off-by: William Douglas <william.douglas@intel.com>
---
 modules/pam_env/pam_env.c | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/modules/pam_env/pam_env.c b/modules/pam_env/pam_env.c
index d2b4cb1..96b3984 100644
--- a/modules/pam_env/pam_env.c
+++ b/modules/pam_env/pam_env.c
@@ -383,7 +383,9 @@ static int read_file(const pam_handle_t *pamh, const char*filename, char ***line
     D(("Parsed file name is: %s", filename));
 
     if ((conf = fopen(filename,"r")) == NULL) {
-      pam_syslog(pamh, LOG_ERR, "Unable to open env file: %s", filename);
+      if (errno != ENOENT) {
+           pam_syslog(pamh, LOG_ERR, "Unable to open env file: %s: %m", filename);
+      }
       return PAM_IGNORE;
     }
 
-- 
2.41.0

