From b6b728ab303f2ac96e8579c30f56f1479280b994 Mon Sep 17 00:00:00 2001
From: William Douglas <william.douglas@intel.com>
Date: Thu, 6 Jul 2023 10:58:31 -0700
Subject: [PATCH] pam_shells: Support a stateless configuration by default

Signed-off-by: Ikey Doherty <michael.i.doherty@intel.com>
Signed-off-by: William Douglas <william.douglas@intel.com>
---
 modules/pam_shells/pam_shells.c | 17 +++++++++++++----
 1 file changed, 13 insertions(+), 4 deletions(-)

diff --git a/modules/pam_shells/pam_shells.c b/modules/pam_shells/pam_shells.c
index abebdd0..2f96c1c 100644
--- a/modules/pam_shells/pam_shells.c
+++ b/modules/pam_shells/pam_shells.c
@@ -26,6 +26,7 @@
 #include <security/pam_ext.h>
 
 #define SHELL_FILE "/etc/shells"
+#define DEFAULT_SHELL_FILE "/usr/share/defaults/etc/shells"
 #define SHELLS "shells"
 #define ETCDIR "/etc"
 #define DEFAULT_SHELL "/bin/sh"
@@ -54,6 +55,7 @@ static int perform_check(pam_handle_t *pamh)
     const char *userName;
     const char *userShell;
     struct passwd * pw;
+    const char *shellPath = NULL;
 
     retval = pam_get_user(pamh, &userName, NULL);
     if (retval != PAM_SUCCESS) {
@@ -109,12 +111,19 @@ static int perform_check(pam_handle_t *pamh)
     char shellFileLine[256];
     FILE * shellFile;
 
-    if (!check_file(SHELL_FILE, pamh))
-        return PAM_AUTH_ERR;
+    if (!check_file(SHELL_FILE, pamh)) {
+        if (!check_file(DEFAULT_SHELL_FILE, pamh)) {
+            return PAM_AUTH_ERR;
+        }
+        shellPath = DEFAULT_SHELL_FILE;
+    } else {
+        /* System administrator has provided /etc/shells */
+        shellPath = SHELL_FILE;
+    }
 
-    shellFile = fopen(SHELL_FILE,"r");
+    shellFile = fopen(shellPath,"r");
     if (shellFile == NULL) {       /* Check that we opened it successfully */
-	pam_syslog(pamh, LOG_ERR, "Error opening %s: %m", SHELL_FILE);
+	pam_syslog(pamh, LOG_ERR, "Error opening %s: %m", shellPath);
 	return PAM_SERVICE_ERR;
     }
 
-- 
2.41.0

