From a582d54cff97f7e2780780eebf6deea4637e321f Mon Sep 17 00:00:00 2001
From: Auke Kok <auke-jan.h.kok@intel.com>
Date: Tue, 26 Nov 2019 11:08:34 -0800
Subject: [PATCH] Allow limits.conf to be missing.

---
 modules/pam_limits/pam_limits.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/modules/pam_limits/pam_limits.c b/modules/pam_limits/pam_limits.c
index 4bc4ae7..d874779 100644
--- a/modules/pam_limits/pam_limits.c
+++ b/modules/pam_limits/pam_limits.c
@@ -750,9 +750,9 @@ parse_config_file(pam_handle_t *pamh, const char *uname, uid_t uid, gid_t gid,
         pam_syslog(pamh, LOG_DEBUG, "reading settings from '%s'", CONF_FILE);
     fil = fopen(CONF_FILE, "r");
     if (fil == NULL) {
-        pam_syslog (pamh, LOG_WARNING,
+        pam_syslog (pamh, LOG_DEBUG,
 		    "cannot read settings from %s: %m", CONF_FILE);
-        return PAM_SERVICE_ERR;
+        return PAM_SUCCESS;
     }
 
     /* start the show */
-- 
2.24.0

